Feature: duplicate event

Victor-Lindh opened the pull request.

gularsson reviewed (commented):
As mentioned on the meeting we should probably add the functionality here:
![Screenshot_2025-03-03-14-25-27_20752](https://github.com/user-attachments/assets/71615ca3-889a-4682-8578-58ef8270366b)
And we should find another place for it in the view, not sure where though.
![2025-03-03_14-31](https://github.com/user-attachments/assets/c8a84353-c55d-4202-96cf-89240891756a)
Händelse-menu could be clickable, we could add a tripple-dot menu to the header, or we could copy the sheet used in editor. Non is perfect though.

gularsson commented on src/components/Duplicate/index.tsx:
```diff
@@ -0,0 +1,89 @@
+            console.log('provider', provider)
+            if (provider && status === 'authenticated' && duplicateId && session) {
+              try {
+                console.log('sending', duplicateId)
```
:broom:

gularsson commented on src/components/Duplicate/index.tsx:
```diff
@@ -0,0 +1,89 @@
+                    }
+                  })
+                )
+                toast.success('Kopieringen lyckades!')
```
Would be nice with a link to the event here

Victor-Lindh replied:
Added new commit that does this

gularsson commented on src/defaults/templates/duplicateTemplate.ts:
```diff
@@ -0,0 +1,104 @@
+    }
+
+    const value: Record<string, BlockType[]> = newsDocDocument[key]
+    console.log('value', value)
```
:broom:

gularsson commented on src/defaults/templates/duplicateTemplate.ts:
Did you try cloning the initial document? I still think it would be more terse and less error prone.

Victor-Lindh replied:
Did try, didn't get it to work

dannelundqvist commented:
> As mentioned on the meeting we should probably add the functionality here: ![Screenshot_2025-03-03-14-25-27_20752]
I would suggest we simply start with moving the button to the header (and maybe then adjust borders and background if needed to make it look nice). Then we can think harder on how to incorporate a dot menu or something else as a pattern for all views.

<img width="936" alt="Skärmavbild 2025-03-03 kl  16 44 29" src="https://github.com/user-attachments/assets/789acc42-2c7a-4ccf-addb-b234329123ee" />

dannelundqvist commented:
I think we should assume things work out as we want it. Something like `Händelsen "Pressträff om ny generaldirektör" kopierades till 2025-03-12` would be more useful to the user and remind the user where the copy went.

When/if we later collect all notifications in a notification center/archive that text would also be more useful than five notifications that says "Kopiering lyckades!". We want to establish a pattern for notification texts.

dannelundqvist commented:
> I think we should assume things work out as we want it. Something like `Händelsen "Pressträff om ny generaldirektör" kopierades till 2025-03-12` would be more useful to the user and remind the user where the copy went.

dannelundqvist commented on src/defaults/templates/duplicateTemplate.ts:
```diff
@@ -0,0 +1,103 @@
+      return ''
+    }
+
+    const [_, time] = date1ISO.split('T')
```
Not that it matters, but there is no real reason to use _ for this case. It could be written as
```typescript
const [, time] = date1ISO.split('T')
```

Victor-Lindh replied:
fixed in new commit

Victor-Lindh commented:
Added new commit that moves it into the view header. Also, replaced the button-esque trigger with a suitable icon instead, to save some space.

And we should find another place for it in the view, not sure where though. ![2025-03-03_14-31](https://private-user-images.githubusercontent.com/8686091/418592958-c8a84353-c55d-4202-96cf-89240891756a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDEwNzgyNjcsIm5iZiI6MTc0MTA3Nzk2NywicGF0aCI6Ii84Njg2MDkxLzQxODU5Mjk1OC1jOGE4NDM1My1jNTVkLTQyMDItOTZjZi04OTI0MDg5MTc1NmEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDMwNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAzMDRUMDg0NjA3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ODIwOWZmNzI1ZWUyMjI2NWZkOTQ4M2ZlN2Y2MzAwMzA0ZWZhZTJhMGM5ZjI0MGVjN2ZjMmZmNTkxOGZiYmIwNCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.09J6By0MWtvUNd4Ao-e5wu3CVjQrAU_FLiigZBJm43s) Händelse-menu could be clickable, we could add a tripple-dot menu to the header, or we could copy the sheet used in editor. Non is perfect though.
